lib/request.js

```diff
@@ -138,20 +138,17 @@ function dispatch (err, data) {
     try {
       const stream = this.factory(data)
       if (stream) {
-        finished(stream, this.callback)
+        finished(stream, (err) => {
+          stream.destroy(err)
```

Comment:
This is a bit of an edge case I'm unsure how to trigger exactly in a test... will need to give it more thought.